vector.js

```diff
@@ -1,212 +1,283 @@
+): $ReadOnlyArray<number> {
+    var points = _.toArray(arguments);
+    var zipped = _.zip.apply(_, points);
+    return zipped.map(arraySum);
+}
+
+export function subtract(
+    v1: $ReadOnlyArray<number>,
+    v2: $ReadOnlyArray<number>
+): $ReadOnlyArray<number> {
+    return _.zip(v1, v2).map(dim => dim[0] - dim[1]);
+}
+
+export function negate(v: $ReadOnlyArray<number>): $ReadOnlyArray<number> {
+    return v.map(x => {
+        return -x;
+    });
+}
+
+// Scale a vector
+export function scale(
+    v1: $ReadOnlyArray<number>,
+    scalar: number
+): $ReadOnlyArray<number> {
+    return v1.map(x => {
+        return x * scalar;
+    });
+}
+
+export function equal(
+    v1: $ReadOnlyArray<number>,
+    v2: $ReadOnlyArray<number>,
+    tolerance?: number
+): boolean {
+    // _.zip will nicely deal with the lengths, going through
+    // the length of the longest vector. knumber.equal then
+    // returns false for any number compared to the undefined
+    // passed in if one of the vectors is shorter.
+    return _.zip(v1, v2).every(pair =>
+        knumber.equal(pair[0], pair[1], tolerance)
+    );
+}
+
+export function codirectional(
+    v1: $ReadOnlyArray<number>,
+    v2: $ReadOnlyArray<number>,
+    tolerance?: number
+): boolean {
+    // The origin is trivially codirectional with all other vectors.
+    // This gives nice semantics for codirectionality between points when
+    // comparing their difference vectors.
+    if (
+        knumber.equal(length(v1), 0, tolerance) ||
+        knumber.equal(length(v2), 0, tolerance)
+    ) {
+        return true;
     }
-};
 
-module.exports = kvector;
+    v1 = normalize(v1);
+    v2 = normalize(v2);
+
+    return equal(v1, v2, tolerance);
+}
+
+export function collinear(
+    v1: $ReadOnlyArray<number>,
+    v2: $ReadOnlyArray<number>,
+    tolerance?: number
+): boolean {
+    return (
+        codirectional(v1, v2, tolerance) ||
+        codirectional(v1, negate(v2), tolerance)
+    );
+}
+
+// Convert a cartesian coordinate into a radian polar coordinate
+export function polarRadFromCart(
+    v: $ReadOnlyArray<number>
+): $ReadOnlyArray<number> {
+    var radius = length(v);
+    var theta = Math.atan2(v[1], v[0]);
+
+    // Convert angle range from [-pi, pi] to [0, 2pi]
+    if (theta < 0) {
+        theta += 2 * Math.PI;
+    }
 
+    return [radius, theta];
+}
+
+// Converts a cartesian coordinate into a degree polar coordinate
+export function polarDegFromCart(
+    v: $ReadOnlyArray<number>
+): $ReadOnlyArray<number> /* TODO: convert to tuple/Point */ {
+    var polar = polarRadFromCart(v);
+    return [polar[0], (polar[1] * 180) / Math.PI];
+}
+
+/* Convert a polar coordinate into a cartesian coordinate
+ *
+ * Examples:
+ * cartFromPolarRad(5, Math.PI)
+ * cartFromPolarRad([5, Math.PI])
+ */
+export function cartFromPolarRad(
+    radius: number | [number, number],
+    theta?: number
+): $ReadOnlyArray<number> /* TODO: convert to tuple/Point */ {
```

Comment:
Is there an issue that's preventing this from being done right now?

Reply:
See earlier comment about `vector`.